### ISSUE TITLE
Bug: Compose View is slow to open

### ISSUE THREAD

**Issue by praveen7557** [InboxSDK/InboxSDK#1098]
**Describe the bug**

When I click on the Compose CTA, the Compose Window opens with a lag. This is happening only recently. We're using compose `addButton` to add some of our components.

**Add any other context about the problem here if anything**

We're using the latest version `@inboxsdk/core@2.1.24  `

**Comment by wegry** [comment]
Hey @praveen7557, thanks for the report. To help us repro, two things would help:

- a code snippet or example extension
- A recording of the slowness you're seeing. Is it 5s, 10s, or 15s for Compose windows opening?

**Comment by praveen7557** [comment]
Hey @wegry,

I attempted to utilize past versions to confirm whether it is indeed InboxSDK. Everything seems to be functioning well up to version 2.1.5; however, beyond that point, there is a noticeable delay in responsiveness. 

In the accompanying video, you can observe that clicking on the Compose CTA results in a slow opening, and even when attempting to enter recipients, the process is sluggish. Conversely, in the same video, after downgrading to version 2.1.5, you can observe a marked improvement in speed and responsiveness.

https://drive.google.com/file/d/1e_zVsRon1AzMGo_r87YgnlgFiKIce0Ul/view?usp=sharing

**Comment by wegry** [comment]
Thanks for the video and snippet. Was there a specific reason `@inboxsdk/sdk@2.1.5` was chosen? Or was that the version you folks were on previously?

**Comment by praveen7557** [comment]
No reason, I was just trying to pinpoint which version had issues, 2.1.5 was the last one where we didn't have issues.

**Comment by praveen7557** [comment]
Hey @wegry, any update here?

**Comment by wegry** [comment]
Hey @praveen7557, contributions welcome here.

A few potential avenues where the slowdown could be coming from:

- Updates around mole placement #1047 that affect ComposeView display in some cases
- Increased DOM logging when ThreadView detection isn't working as expected #1079

**Comment by praveen7557** [comment]
Hey @wegry, I went through what changed from 2.1.5 to 2.1.6, [this](https://github.com/InboxSDK/InboxSDK/commit/51e30c080f0e1b15cd417c0fc20b26645ac77145) commit is the one that's the cause of the issue. I'm not sure what the functionality of it is. 

For now, I just updated the selectors in `2.1.5` and using it for my use case. Please, let me know if this can be fixed.

**Comment by wegry** [comment]
Thanks for digging in @praveen7557. Just to confirm: do you have a diff of the selector change you have locally that you'd be willing to share?

If you mean the selector [here](https://github.com/InboxSDK/InboxSDK/pull/1043/files#diff-4deb24efbdddeae857792fd63a7c2937123ae6c6f6863bd2bfbe6ccacdeaad88R481), we use that to respond to the side panel changing size.

Note the bottom right corner of these two screenshots:

![Screenshot 2023-12-18 at 09 05 48](https://github.com/InboxSDK/InboxSDK/assets/5156873/bd1a3e1a-70c8-4158-90b1-672df91863d5)
![Screenshot 2023-12-18 at 09 05 34](https://github.com/InboxSDK/InboxSDK/assets/5156873/14d06058-cc70-4e13-a9d0-e7d101d6ade2)

**Comment by praveen7557** [comment]
Hey @wegry, sorry for the delay in replying. I was using 2.1.5 with a few updated selectors, you can see my commit here https://github.com/praveen7557/InboxSDK/commit/46fb0689c0bee6df7134c29901307ac318c0507b. I tried using the latest version `2.1.38` but it is still slow.

I ran the extension in the performance tab once with my changes `2.1.5` version and once with `2.1.38` version, you can see the difference below 

2.1.5

<img width="1431" alt="Screenshot 2024-04-03 at 1 27 56 PM" src="https://github.com/InboxSDK/InboxSDK/assets/13882201/b560cf71-5121-4335-b815-1afcf895726a">

2.1.38

<img width="1426" alt="Screenshot 2024-04-03 at 1 27 28 PM" src="https://github.com/InboxSDK/InboxSDK/assets/13882201/3d3455a6-e47f-44d2-b4ce-8777e1075113">


the `flush` call which is taking around 800ms is from InboxSDK. Also, we're getting this error message `[Violation] Forced reflow while executing JavaScript took 1458ms` when using this.

**Comment by wegry** [comment]
@praveen7557 thanks for the profiling. Do we know what the stack around `flush` looks like? Source maps might help a great deal.

**Comment by praveen7557** [comment]
again, to reitarate one of my first comments, slowness started after this commit https://github.com/InboxSDK/InboxSDK/commit/51e30c080f0e1b15cd417c0fc20b26645ac77145

**Comment by wegry** [comment]
That looks an awful lot like an `asap` library call then. `asap`'s roughly equivalent to the modern `queueMicrotask`.

Is this `flush` call issue something you can reproduce with no other extensions enabled?

**Comment by praveen7557** [comment]
yes, it's reproducible, just tried in incognito with just my extension enabled, it is slow.

**Comment by wegry** [comment]
With the following example extension (borrowed from above but without the nested React rendering):

```js
InboxSDK.load(2, 'custom-thread-list').then((sdk) => {
  if (!document.querySelector('#GmailSidebar')) {
    const el = document.createElement('div');
    el.id = 'GmailSidebar';
    el.textContent = 'Hello, World!';

    sdk.Global.addSidebarContentPanel({
      el,
      title: 'Teddy',
      iconUrl:
        'https://img.freepik.com/premium-vector/cute-teddy-bear-cartoon-icon_192760-116.jpg?w=2000',
    }).then((contentPanel) => {
      contentPanel.close();
      // 3). then((contentPanel) → {
      //   contentPanel. close(); setPanel (contentPanel);
      // }
      // );
    });
  }
});
```

I'm still not able to reproduce slowness with the compose opening. Vanilla Gmail without extensions flags `[violation] 'setTimeout handler took X'`. That's not to say there isn't an issue here. It could be account specific, for example.

**Comment by praveen7557** [comment]
can you try adding something in the compose view also, we're adding some buttons in the compose window 
`inboxSdk.Compose.registerComposeViewHandler(initComposeWindowBasics)`.

`[Violation] Forced reflow while executing JavaScript took 1458ms` we're getting these kind of violation also only with this extension enabled.

**Comment by wegry** [comment]
Can you add something to the [example](https://github.com/InboxSDK/InboxSDK/issues/1098#issuecomment-2034655411) here similar to your usage? Maybe there's something creeping in there? 

Testing with the example extension we have access to would be extremely helpful.

**Comment by M-J-Murray** [comment]
@praveen7557 did you manage to figure out what was causing the slow compose view load? I'm having the same issue.

**Comment by M-J-Murray** [comment]
Here is a really simple example that produces a lag for me. When gmail has fully loaded my inbox and I open a compose view, it can take a few seconds for the console to log that the compose view has appeared and to add the button:
```javascript
InboxSDK.load(2, "...").then((sdk) => {
	console.log("Loaded inbox sdk");
	sdk.Compose.registerComposeViewHandler((composeView) => {
		console.log("Loaded compose view;");
		composeView.addButton({title: "Test"});
	});
});
```
I'm using the manifest V3 version of InboxSDK.

**Comment by M-J-Murray** [comment]
I just checked the https://github.com/InboxSDK/hello-world/tree/main repo and it has the same slow issue:

https://github.com/user-attachments/assets/a0843e0f-24f6-4326-985e-715049675a62
